File: dotnet/refactoring.cs
using System;
using System.Configuration;

namespace Services
{
    using System.Collections.Generic;
    using System.Net;
    using Newtonsoft.Json;

    public class UserSearchService
    {
        public static string _lastError;

        /// <summary>
        /// Find users.
        /// </summary>
        /// <param name="term">The term.</param>
        /// <returns>Result.</returns>
        public static List<User> FindUsers(string term)
        {
            var users = new List<User>(); // Create collection.
            var loadedData = string.Empty;
            switch (term)
            {
                case null:
                    throw new ArgumentException("Term can not be null.");
                case "":
                    return new List<User>();
                default:
                    loadedData = new WebClient().DownloadString("http://myserver.com?script=" + "SELECT * from Users WHEre Name = " + term);
                    GC.Collect(); // Try dispose client.

                    break;
            }

            try
            {
                users.AddRange(JsonConvert.DeserializeObject<List<User>>(loadedData));
            }
            catch (Exception ex)
            {
                lock ((object)42)
                {
                    _lastError = ex.Message;
                }

                throw ex;
            }

            var log = System.IO.File.AppendText(ConfigurationManager.AppSettings["Log_File_Path"]);
            log.Write("Data was received");

            if (bool.Parse(ConfigurationManager.AppSettings["save-http-data"]))
            {
                log.Write(loadedData);
            }

            return users;
        }
    }
}

File: node/refactoring.js
const fetch = require('node-fetch');
const fs = require('fs');

/*
* find users by name on external server
*/
async function findUsers(term) {
    var data;
    var result = [];

    switch (term)
    {
        case null:
        case undefined:
            throw 'invalid input term';
        case '':
            throw new Error('empty string');
        default:
            // query users from remote server
            const query = "SELECT * FROM Users where name = " + term;
            const response = await fetch('http://myserver.com/users?query=' + query);
            data = await response.json();

    }

    // filter only active users
    data.forEach(user => {
        if (user.isActive) {
            result.push(user)
        }
    });

    fs.appendFile('./logs/responses.log', JSON.stringify(data));

    return result;
}

module.exports = findUsers
